cmd/mapcidr/main.go

```diff
@@ -318,32 +335,23 @@ func process(wg *sync.WaitGroup, chancidr, outputchan chan string) {
 		if options.Aggregate || options.Shuffle || hasSort || options.AggregateApprox || options.Count {
 			_ = ranger.AddIPNet(pCidr)
 			allCidrs = append(allCidrs, pCidr)
-		} else if options.Slices > 0 {
-			subnets, err := mapcidr.SplitN(cidr, options.Slices)
-			if err != nil {
-				gologger.Fatal().Msgf("%s\n", err)
-			}
-			for _, subnet := range subnets {
-				outputchan <- subnet.String()
-			}
-		} else if options.HostCount > 0 {
-			subnets, err := mapcidr.SplitByNumber(cidr, options.HostCount)
-			if err != nil {
-				gologger.Fatal().Msgf("%s\n", err)
-			}
-			for _, subnet := range subnets {
-				outputchan <- subnet.String()
-			}
 		} else {
-			var ipFlagList []string
-			ipFlagList = append(ipFlagList, prepareIPsFromCidrFlagList(options.MatchIP)...)
-			ipFlagList = append(ipFlagList, prepareIPsFromCidrFlagList(options.FilterIP)...)
-			ips, err := mapcidr.IPAddressesAsStream(cidr)
+			commonFunc(cidr, outputchan)
+		}
+	}
+	if options.IPRangeInput {
```

Comment:
`options.IPRangeInput` is not needed. We can directly iterate on `ipRangeList` (it will be skipped if empty)